src/lib/y2partitioner/widgets/fstab_options.rb

```diff
@@ -0,0 +1,538 @@
+      end
+
+      # No all the fstab options are supported by all the filesystem so each
+      # widget are able to check if the current filesystem is supported
+      # explicitely or checking if the values it is responsable of are
+      # supported by the filesystem.
+      def supported_by_filesystem?
+        return false if !@options.filesystem_type
+
+        if self.class.const_defined?("SUPPORTED_FILESYSTEMS")
+          self.class::SUPPORTED_FILESYSTEMS
+            .include?(@options.filesystem_type.to_sym)
+        else
+          self.class::VALUES.all? do |v|
+            @options.filesystem_type.supported_fstab_options.include?(v)
+          end
+        end
+      end
+
+      # @param widget [CWM::AbstractWidget]
+      # @return [Boolean] true if it is supported by current filesystem
+      def ui_term?(widget)
+        return true if widget.supported_by_filesystem?
+
+        false
+      end
+
+      # @param widget [CWM::AbstractWidget]
+      # @return [CWM::WidgetTerm]
+      def to_ui_term(widget)
+        return Empty() unless ui_term?(widget)
+
+        Left(widget)
+      end
+
+      # @param widget [CWM::AbstractWidget]
+      # @return [Array<CWM::WidgetTerm>]
+      def ui_term_with_vspace(widget)
+        return [Empty()] unless ui_term?(widget)
+
+        [Left(widget), VSpacing(1)]
+      end
+
+      def delete_from_fstab!(option)
+        @options.fstab_options.delete_if { |o| o =~ option }
+      end
+
+    private
+
+      # Common regexp checkbox widgets init.
+      def init_regexp
+        i = @options.fstab_options.index { |o| o =~ self.class::REGEXP }
+
+        self.value =
+          if i
+            @options.fstab_options[i].gsub(self.class::REGEXP, "")
+          else
+            self.class::DEFAULT
+          end
+      end
+    end
+
+    # Push button that launch a dialog for set the fstab options
+    class FstabOptionsButton < CWM::PushButton
+      include FstabCommon
+
+      def label
+        _("Fstab options...")
+      end
+
+      def handle
+        Dialogs::FstabOptions.new(@options).run
+
+        nil
+      end
+    end
+
+    # FIXME: The help handle does not work without wizard
+    # Main widget for set all the available options for a particular filesystem
+    class FstabOptions < CWM::CustomWidget
+      include FstabCommon
+
+      SUPPORTED_FILESYSTEMS = %i(btrfs ext2 ext3 ext4 reiserfs).freeze
+
+      def initialize(options)
+        @options = options
+
+        self.handle_all_events = true
+      end
+
+      def init
+        disable if !supported_by_filesystem?
+      end
+
+      def handle(event)
+        case event["ID"]
+        when :help
+          help = []
+
+          widgets.each do |w|
+            help << w.help if w.respond_to? "help"
+          end
+
+          Yast::Wizard.ShowHelp(help.join("\n"))
+        end
+
+        nil
+      end
+
+      def contents
+        VBox(
+          Left(MountBy.new(@options)),
```

Comment:
I checked the rendered dialog, and a `VSpacing(1)`  is missing between MountBy and VolumeLabel.
(The legacy dialog also has this bug. IMHO we should fix this.)